### ISSUE TITLE
Add-ons: add-on details are missing from the order details screens

### ISSUE THREAD

**Issue by designsimply** [woocommerce/woocommerce-ios#4271]
To prepare:

1. Have a site with [Product Add-ons](https://woocommerce.com/products/product-add-ons/) installed.
1. On the web, create a new order including an add-on.
1. In the app, go to Settings > Experimental Features and toggle the order add-ons switch on.
1. Close and re-open the app.

To test:

1. Go to orders > mark the order as complete or put it on hold, and tap "Details" under the products section.
1. Look for add-on information on the order details product list screen.

**Result:** orders containing products with add-ons applied to no show add-on information on the order details screen. ([2m52s](https://cloudup.com/cKXHUoY2jFq))

Order|Order > Products > Details|View Add-Ons|Web-view of the same order
---|---|---|---
<img width="470" alt="Screen Shot 2021-05-24 at May 24 7 11 57 PM" src="https://user-images.githubusercontent.com/1119271/119425084-f2052480-bcc3-11eb-8800-c99bd599b1ce.png">|<img width="470" alt="Screen Shot 2021-05-24 at May 24 6 54 04 PM" src="https://user-images.githubusercontent.com/1119271/119424037-e44e9f80-bcc1-11eb-8aee-b18505ecef39.png">|<img width="470" alt="Screen Shot 2021-05-24 at May 24 6 53 55 PM" src="https://user-images.githubusercontent.com/1119271/119424051-eca6da80-bcc1-11eb-925f-98161beb9ffe.png">|<img width="470" alt="Screen Shot 2021-05-24 at May 24 6 56 43 PM" src="https://user-images.githubusercontent.com/1119271/119424073-f8929c80-bcc1-11eb-816e-2d658513678f.png">

**Note:** this happened for me regardless of whether the order contained add-on products grouped or not.

Order|Order > Product > Details|View Add-Ons|Web-view of the same order
---|---|---|---
![IMG_4145](https://user-images.githubusercontent.com/1119271/119425657-08f84680-bcc5-11eb-8d95-d53129137ce0.PNG)|![IMG_4147](https://user-images.githubusercontent.com/1119271/119425650-07c71980-bcc5-11eb-84fe-fde733243dba.PNG)|![IMG_4146](https://user-images.githubusercontent.com/1119271/119425653-085fb000-bcc5-11eb-95e0-ddcf57feda51.PNG)|<img width="1440" alt="Screen Shot 2021-05-24 at May 24 7 18 40 PM" src="https://user-images.githubusercontent.com/1119271/119425586-e8c88780-bcc4-11eb-98bf-9cd18294bea6.png">

<sup>(internal reference: p5T066-2iC-p2#comment-8307 /ht @astralbodies)</sup>

**Mobile Environment**

- Device: iPhone 6S
- iOS version: 14.4.2
- WooCommerce iOS version: 6.7.0.0 (TestFlight beta)

**WordPress Environment**
<details>

```
### WordPress Environment ###

WordPress address (URL): https://double-reef.jurassic.ninja
Site address (URL): https://double-reef.jurassic.ninja
WC Version: 5.3.0
REST API Version: ✔ 5.3.0
WC Blocks Version: ✔ 4.9.1
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 2.2.6
Log Directory Writable: ✔
WP Version: ❌ 5.7.2 - There is a newer version of WordPress available (5.7.3-alpha-50937)
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.46 (Unix) OpenSSL/1.0.2g
PHP Version: 7.4.18
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.33-0ubuntu0.16.04.1-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.3.0
WC Database Prefix: wp_
Total Database Size: 13.03MB
Database Data Size: 10.02MB
Database Index Size: 3.01MB
wp_woocommerce_sessions: Data: 0.03MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.08MB + Index: 0.09MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.03MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ce4wp_abandoned_checkout: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.05MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_custom_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_dynamic_segment_filters: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_feature_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_mapping_to_external_entities: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_newsletter_links: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_newsletter_option: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_option_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_posts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_segment: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_templates: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_scheduled_tasks: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_scheduled_task_subscribers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_segments: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_sending_queues: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_settings: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_statistics_clicks: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_statistics_forms: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_statistics_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_statistics_opens: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_mailpoet_statistics_unsubscribes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_statistics_woocommerce_purchases: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_mailpoet_stats_notifications: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_subscribers: Data: 0.02MB + Index: 0.13MB + Engine InnoDB
wp_mailpoet_subscriber_custom_field: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_subscriber_ips: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_subscriber_segment: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_user_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 4.11MB + Index: 0.19MB + Engine InnoDB
wp_postmeta: Data: 1.52MB + Index: 0.48MB + Engine InnoDB
wp_posts: Data: 0.25MB + Index: 0.09MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.05MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 54
global_product_addon: 1
mailpoet_page: 1
nav_menu_item: 102
page: 21
post: 40
product: 16
product_variation: 220
shop_order: 24
shop_order_refund: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (13) ###

Companion Plugin: by Osk – 1.18
Creative Mail by Constant Contact: by Constant Contact – 1.3.8
Facebook for WooCommerce: by Facebook – 2.5.0
Jetpack by WordPress.com: by Automattic – 9.7
Google Ads & Marketing by Kliken: by Kliken – 1.0.7
MailPoet 3 (New): by MailPoet – 3.61.0
WooCommerce Smooth Generator: by Automattic – 1.0.1
WCS staging: by mb – 1.0
WooCommerce Payments: by Automattic – 2.4.0
WooCommerce Product Add-ons: by WooCommerce – 4.0.0
WooCommerce Shipping & Tax: by WooCommerce – 1.25.13
WooCommerce: by Automattic – 5.3.0
WordPress Beta Tester: by Peter Westwood
Andy Fragen – 3.1.0


### Inactive Plugins (1) ###

Akismet Anti-Spam: by Automattic – 4.1.9

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #1364 - /shop/
Cart: #1365 - /cart/
Checkout: #1366 - /checkout/
My account: #1367 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 3.6.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 62
Oldest: 2021-05-17 21:37:45 +0000
Newest: 2021-05-25 00:25:19 +0000


### Status report information ###

Generated at: 2021-05-25 00:43:14 +00:00
```
</details>

**Comment by Ecarrion** [comment]
Hi @designsimply! I'm not totally sure I understand the bug.

Is it that when you mark the order as "complete" and then tap on `details` you see a list of products but no information about their add-ons?

If that is the case, I think that would be a feature request!

Also, what would you want to see there? The "view add-ons" button? or all the product attributes information?

cc @adamzelinski

**Comment by designsimply** [comment]
As a user, I expect to be able to see all the details for the products in an order to be available on any details-related screens for the order. 

For example, in Order 1651, I ordered 1 test product with a Gift Wrapping add-on and 1 test product with an Engraving add-on. Here's how the order looked on the web when I placed it:

<img width="1440" alt="Screen Shot 2021-05-25 at May 25 10 42 59 AM" src="https://user-images.githubusercontent.com/1119271/119539385-5918ec80-bd49-11eb-95ec-6db7f28daa15.png">

In the app, when I check the order while the status is on hold or completed, the main order screen shows 2 test products and mentions gift wrapping but not the engraving:

<img src="https://user-images.githubusercontent.com/1119271/119539534-82397d00-bd49-11eb-8076-6b9f7feb2cde.PNG" width="270" alt="IMG_4151">

When I tap View Add-Ons, I can see one charge for gift wrapping but the detail about the engraving add-on is missing: 

<img src="https://user-images.githubusercontent.com/1119271/119539667-b1e88500-bd49-11eb-9d2b-a1f637f2d8cf.PNG" width="270" alt="IMG_4153">

When I tap Details, I can see that 2 test products were ordered but there is not any information about add-ons at all: 

<img src="https://user-images.githubusercontent.com/1119271/119539737-caf13600-bd49-11eb-98c4-cfb919f5cc11.PNG" width="270" alt="IMG_4152">

In Order 1651, I cannot tell at all that one of the test products has an engraving add-on and I think that's a bug. It's possible this may be covered by the grouped products by different add-ons issue https://github.com/woocommerce/woocommerce-ios/issues/4126, however, some add-on details only ever show up on the main order screen.

For example, in Order 1653, I ordered one product with an engraving add-on, and no "View Add-on" option is presented and the Details screen also doesn't mention engraving or that the order has any add-ons:

<img src="https://user-images.githubusercontent.com/1119271/119541546-c2016400-bd4b-11eb-8484-209a90207c34.PNG" width="270" alt="IMG_4156"> <img src="https://user-images.githubusercontent.com/1119271/119541542-c168cd80-bd4b-11eb-94a9-fb75b35fe2ef.PNG" width="270" alt="IMG_4157">

Am I thinking about Details incorrectly? Is that really a summary product list and details are not expected there? Why isn't "View Add-on" presented on the order screen for a product with a text field add-on?

**Comment by Ecarrion** [comment]
Hi @designsimply 

Any issue related to an order that has two of the same products will be fixed by #4126. Unfortunately we are blocked by p91TBi-566-p2#comment-4513

----

I will also make sure that the "view add-ons" button is visible no matter what status the order has.

----

For

> For example, in Order 1653, I ordered one product with an engraving add-on, and no "View Add-on" option is presented and the Details screen also doesn't mention engraving or that the order has any add-ons:

<img width="284" alt="Screen Shot 2021-05-25 at 4 14 50 PM" src="https://user-images.githubusercontent.com/562080/119569437-5aabda00-bd74-11eb-9163-44ce28405314.png">

For the order product detail screen, I missed covering that case. 
@adamzelinski what do you think we should do there? Should we show a "view add-on" button in that cell or is it enough with the "view add-ons" button from the order detail screen?

**Comment by ghost** [comment]
> For the order product detail screen, I missed covering that case.
> @adamzelinski what do you think we should do there? Should we show a "view add-on" button in that cell or is it enough with the "view add-ons" button from the order detail screen?

I think for what was in scope with the workaround we could just show the 'view add-ons' button here thanks @Ecarrion

**Comment by Ecarrion** [comment]
Hi @designsimply,

Regarding the issue where orders with on-hold status weren't showing the "view add-ons" button. Could you confirm that you had the beta-feature enabled? 

I just tested it using your credentials and I can see the "view add-on" button correctly!

<img width="431" alt="Screen Shot 2021-05-26 at 5 17 41 PM" src="https://user-images.githubusercontent.com/562080/119738734-c3fc1d80-be46-11eb-9832-bdae14ffe9e6.png">

**Comment by designsimply** [comment]
> Regarding the issue where orders with on-hold status weren't showing the "view add-ons" button. Could you confirm that you had the beta-feature enabled?

@Ecarrion, my apologies, let me try to clarify. I could see the "View Add-Ons" button on orders where products were not grouped, like in order 1653. Where I could not see the engraving add-on was in an order with grouped products that had different add-ons, such as in order 1651. 

Here's how order 1651 looks in my WP Admin:

![image](https://user-images.githubusercontent.com/1119271/119897083-27488700-befd-11eb-88ab-4f616205fcb8.png)

And here's how the same order 1651 looks in the app right now after I double checked that the experimental feature toggle is on: 

<img src="https://user-images.githubusercontent.com/1119271/119897647-f4eb5980-befd-11eb-949c-67442d6ba169.png" width="270" alt=""> <img src="https://user-images.githubusercontent.com/1119271/119897668-fcaafe00-befd-11eb-817d-977a75e3b3a0.png" width="270" alt="">

Engraving is never mentioned, however, I think this is part of the grouped items issue that will be fixed as part of #4126 so I can help re-test this after we're able to get a fix for the grouping issue.

**Comment by Ecarrion** [comment]
Hey @designsimply! Oh ok, there is no issue with the status of the order. 
As you mention #4126 will fix all the issues where items are grouped.

Also, #4293 adds the "View Add-on" button in the product detail screen. SO I think this can be closed now!
